File: benchmarks/tsb/bench_us_federal_holidays.ts
/**
 * Benchmark: USFederalHolidayCalendar.holidays() over a 10-year range
 */
import { USFederalHolidayCalendar } from "../../src/index.js";

const WARMUP = 5;
const ITERATIONS = 20;

const start_date = new Date("2000-01-01");
const end_date = new Date("2009-12-31");

for (let i = 0; i < WARMUP; i++) {
  const cal = new USFederalHolidayCalendar();
  cal.holidays(start_date, end_date);
}

const t0 = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  const cal = new USFederalHolidayCalendar();
  cal.holidays(start_date, end_date);
}
const total = performance.now() - t0;

console.log(
  JSON.stringify({
    function: "us_federal_holidays",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_us_federal_holidays.py
"""
Benchmark: pandas USFederalHolidayCalendar.holidays() over a 10-year range
"""
import json
import time
import pandas as pd
from pandas.tseries.holiday import USFederalHolidayCalendar

WARMUP = 5
ITERATIONS = 20

start_date = "2000-01-01"
end_date = "2009-12-31"

for _ in range(WARMUP):
    cal = USFederalHolidayCalendar()
    cal.holidays(start_date, end_date)

t0 = time.perf_counter()
for _ in range(ITERATIONS):
    cal = USFederalHolidayCalendar()
    cal.holidays(start_date, end_date)
total = (time.perf_counter() - t0) * 1000  # ms

print(json.dumps({
    "function": "us_federal_holidays",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
